bootstrap/handlers/secret/client/vaultclient.go

```diff
@@ -0,0 +1,91 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/config"
+	"github.com/edgexfoundry/go-mod-bootstrap/di"
```

Comment:
Do not pass dependency injection container into this package.  Makes testing more complex.  Pass in explicit dependencies retrieved from container by caller.

Reply:
The main reason I did that is because I do not want to introduce the import dependency on the go-mod-core-contract if using the dependencies retrieved from container by caller.  You can argue, yea, that is the container's dependency but that is from the implementation side of container.

Reply:
I'm not willing to compromise on this.  Passing around a god-object (which is what the DIC is) is not desired and makes testing much more complex.  When it was implemented as part of the bootstrap, the intention with DIC was to provide a service locator for dependencies that needed to be referenced (but not used) before they were actually defined.

What's your issue on introducing the import dependency on `go-mod-core-contract`?

Reply:
ok, my original concern is that there is no dependency exists for that in the `secret.go` where the caller is using DIC so I just follow that.  If you are fine with the go-mod-core-contracts on this package, I am fine with that then.

Reply:
go-mod-bootstrap takes a dependency on go-mod-core-contracts already:  https://github.com/edgexfoundry/go-mod-bootstrap/blob/1a682be3a0a69810c96c1e4f87d69a86ffdbb474/go.mod#L6.

Reply:
changed